shapeshift/web
# fix: qrcode camera access declined error handling

## Description

If you refused the camera access permission on your browser, the QR code button is here but it redirects you to the next step instead of telling you that the app doesn't have the access right to the camera

## Notice

Before submitting a pull request, please make sure you have answered the following:


## Pull Request Type


## Issue (if applicable)

closes #935

## Risk

QrReader component is just sending browser strings as an error, I don't really know how to handle every errors with this.

If there are some risks, it's on the QR Code Reader feature when you try to send a TX.

The `QRCodeScanner` component is displaying a message so you understand why you've an error, and permit you to click on "Back" to be redirected to the address form

## Testing

- Try to send an asset (ETH for example)
- Click on the QR code
- Reject the permission access
- You should be redirected to an error modal and have the possibility to go back
- Try to click on the permissions button, it shouldn't show the QRCode if you didn't give the permissions
- Give the permissions:
If you are on chrome, you can do it here: 
![image](https://user-images.githubusercontent.com/14963751/156630841-3e4e9f81-7127-4d13-9e61-b0746722fada.png)
- Then click on the button, you should be able to see the QRCode reader

## Screenshots (if applicable)


![image](https://user-images.githubusercontent.com/14963751/156658013-4d37dec9-62db-42b8-9a88-69387eac131d.png)

In case the error is not a `Permission denied` error:

![image](https://user-images.githubusercontent.com/14963751/156643431-3dd94a9d-dc86-442c-a2d4-ffdd44e98678.png)
